x-pack/plugin/core/src/main/java/org/elasticsearch/protocol/xpack/graph/GraphExploreRequest.java

```diff
@@ -96,37 +95,14 @@ public GraphExploreRequest indicesOptions(IndicesOptions indicesOptions) {
         return this;
     }
 
-    /**
-     * The document types to execute the explore against. Defaults to be executed against
-     * all types.
-     *
-     * @deprecated Types are in the process of being removed. Instead of using a type, prefer to
-     * filter on a field on the document.
-     */
-    @Deprecated
-    public String[] types() {
-        return this.types;
-    }
-
-    /**
-     * The document types to execute the explore request against. Defaults to be executed against
-     * all types.
-     *
-     * @deprecated Types are in the process of being removed. Instead of using a type, prefer to
-     * filter on a field on the document.
-     */
-    @Deprecated
-    public GraphExploreRequest types(String... types) {
-        this.types = types;
-        return this;
-    }
-
     public GraphExploreRequest(StreamInput in) throws IOException {
         super(in);
 
         indices = in.readStringArray();
         indicesOptions = IndicesOptions.readIndicesOptions(in);
-        types = in.readStringArray();
+        if (in.getVersion().before(Version.V_8_0_0)) {
```

Comment:
I mentioned this in the PR for the explain API as well -- I think it'd be nice to read the types array and assert that it is empty.

Reply:
++